app/controllers/optimizations_controller.rb

```diff
@@ -26,19 +26,13 @@ def select
   def show
   end
 
-  def download
-    authorize Optimization.find(params[:optimization_id])
-    path = "#{Rails.root}/log/optimizations/optim_#{params[:optimization_id]}.log"
-    if File.exist?(path) 
-      send_file(path) 
-    else
-      redirect_to optimizations_url, notice: "There isn't a log file"
-    end
-  end
-
   def new
     @optimization = Optimization.new
     authorize @optimization
+    optim_num = Optimization.owned_by(current_user).size
+    if optim_num > 19
```

Comment:
Avoid the magic-number effect by using a constant `MAX_OPTIM_NUMBER = 20` (is it not 20 for allowing 20 optims?). Ideally this constant should be defined in `Optimization` class as it has to be used also in the `api\v1\optimization_controller.rb.` By the way, could you also implement the optim limit in that file for the JSON REST api? May be you can even add a test for this: create 20 optims and check for an error when trying to create the 21fst.